index.js

```diff
@@ -186,6 +192,9 @@ Router.prototype.handle = function handle(req, res, callback) {
   req.baseUrl = parentUrl
   req.originalUrl = req.originalUrl || req.url
 
+  // trigger the "beginning of route handling" event
+  if (events && routerEvents[0] in events) self.emit('handlestart', req)
```

Comment:
Agreed. I decided on this to keep the naming convention "flexible". Will come up with an better alternative.